File: src/main/java/me/pcasaes/hexoids/core/application/eventhandlers/ApplicationConsumers.java
package me.pcasaes.hexoids.core.application.eventhandlers;

import me.pcasaes.hexoids.core.domain.model.DomainEvent;

import java.util.function.BooleanSupplier;

public interface ApplicationConsumers {


     void onJoinGame(DomainEvent domainEvent);

     void onPlayerAction(DomainEvent domainEvent);

     void onBoltLifeCycle(DomainEvent domainEvent) ;

     void onBoltAction(DomainEvent domainEvent);

     void onScoreBoardControl(DomainEvent domainEvent);

     void onScoreBoardUpdate(DomainEvent domainEvent);

    interface HaveStarted extends BooleanSupplier {
    }

}

File: src/main/kotlin/me/pcasaes/hexoids/core/application/eventhandlers/ApplicationConsumers.kt
package me.pcasaes.hexoids.core.application.eventhandlers

import me.pcasaes.hexoids.core.domain.model.DomainEvent
import java.util.function.BooleanSupplier

interface ApplicationConsumers {
    fun onJoinGame(domainEvent: DomainEvent)

    fun onPlayerAction(domainEvent: DomainEvent)

    fun onBoltLifeCycle(domainEvent: DomainEvent)

    fun onBoltAction(domainEvent: DomainEvent)

    fun onScoreBoardControl(domainEvent: DomainEvent)

    fun onScoreBoardUpdate(domainEvent: DomainEvent)

    fun interface HaveStarted : BooleanSupplier
}
